### PR TITLE
OHARA-1475 Set a maximum length for cluster name

### PR DESCRIPTION
**Pull request by eechih**
Create cluster，cluster name 最多 30 個字元

### CODE REVIEW

File: ohara-manager/client/src/components/pages/Services/WorkerNewModal/WorkerNewModal.js [hunk 1]
```diff
@@ -72,13 +73,18 @@ class WorkerNewModal extends React.Component {
...
    */
-  formatName = value =>
-    value === undefined ? '' : value.replace(/[^0-9a-zA-Z]/g, '');
+  formatName = value => {
+    if (!value) return '';
+    return truncate(value.replace(/[^0-9a-zA-Z]/g, ''), {
+      length: 30,
+      omission: '',
+    });
+  };
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/Services/WorkerNewModal/WorkerNewModal.js]: Does this mean that user can type more than 30 English letters but the name will eventually be truncated by our code?

Comment (konekoya) [reply ohara-manager/client/src/components/pages/Services/WorkerNewModal/WorkerNewModal.js]: Hmm, i need to test this in my local

**Comment by konekoya** [comment]
@eechih I think adding some hints for the users would be helpful. Maybe something like a warning right below the input?

**Comment by konekoya** [comment]
While we're discussing UI... @vitojeng

File: ohara-manager/client/src/components/pages/Services/WorkerNewModal/WorkerNewModal.js (hunk 1 above)
Comment (eechih) [reply ohara-manager/client/src/components/pages/Services/WorkerNewModal/WorkerNewModal.js]: 沒錯，就像你想的那樣。類似在 input 加上 'maxLength' attribute 一樣的效果。

**Comment by vitojeng** [comment]
> While we're discussing UI... @vitojeng

if this workload is not heavy for @eechih, let's us adding it.
Otherwise, we can move it to **v0.3**.

File: ohara-manager/client/src/components/common/Form/Label.js [hunk 2]
```diff
@@ -29,21 +32,48 @@ const LabelWrapper = styled.label`
...
+  position: absolute;
+  right: -16px;
+  top: -2px;
+  font-size: 7px;
+`;
+
 const Label = ({
   children,
   css = { margin: '0 0 8px', width: 'auto' },
+  tooltipId = uuid(),
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Form/Label.js]: Why we need an uuid here?

Comment (konekoya) [reply ohara-manager/client/src/components/common/Form/Label.js]: Is that a requirement from using `react-tooltip` @@?

Comment (eechih) [reply ohara-manager/client/src/components/common/Form/Label.js]: 這是 react-tooltip 的要求。避免 tooltip 之間互相影響。

Comment (konekoya) [reply ohara-manager/client/src/components/common/Form/Label.js]: Thinking about the same thing. Good to know that, thanks!

File: ohara-manager/client/src/components/common/Form/Label.js [hunk 3]
```diff
@@ -29,21 +32,48 @@ const LabelWrapper = styled.label`
 
 LabelWrapper.displayName = 'Label';
 
+const TooltipIcon = styled.span`
+  position: absolute;
+  right: -16px;
+  top: -2px;
+  font-size: 7px;
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Form/Label.js]: Can we make the font size a bit larger? The current font size is too small :)

File: ohara-manager/client/src/components/common/Form/Label.js [hunk 4]
```diff
@@ -29,21 +32,48 @@ const LabelWrapper = styled.label`
...
+  tooltipString,
+  tooltipRender,
   ...rest
 }) => {
   return (
     <LabelWrapper {...css} {...rest}>
       {children}
+      {tooltipString && (
+        <TooltipIcon data-tip={tooltipString}>
+          <i className="fas fa-exclamation-circle" />
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Form/Label.js]: I think we should use an info icon like: https://fontawesome.com/icons/info-circle?style=solid

The current one could potentially confuse users... Not sure if that's just me 😅

File: ohara-manager/client/src/components/common/Form/Label.js (hunk 3 above)
Comment (eechih) [reply ohara-manager/client/src/components/common/Form/Label.js]: 沒問題，調整成與 Label 的 font-size 一樣大

File: ohara-manager/client/src/components/common/Form/Label.js (hunk 4 above)
Comment (eechih) [reply ohara-manager/client/src/components/common/Form/Label.js]: No problem. It has been changed to fa-info-circle.

**Comment by konekoya** [comment]
Looks pretty good now. As I would probably be making some changes throughout our frontend code. To avoid unnecessary merge conflicts. I'm merging this PR. Thank you and happy lunar new year 👍

**Pull request merged**